### ISSUE TITLE
logs show "verify ContainerStatus failed" 

### ISSUE THREAD

**Issue by Jan-EC** [kubernetes/kubernetes#105332]
#### What happened:

kubelet "verify ContainerStatus failed", logs are flooded with this entires

#### What you expected to happen:

running without unknown errors in the logs

#### How to reproduce it (as minimally and precisely as possible):

I just setup a k8s cluster with kubeadm - initially 1.20 and upgraded to 1.21 and 1.22.x using cri-o as container runtime, cgroups from systemd for k8s and cri-o. 

#### Anything else we need to know?:

I tried googling the error - but only found the source code where it is logged and I have no idea what's going on and in which direction to investigate.

#### Environment:
- Kubernetes version (use `kubectl version`): 1.22.1
- Cloud provider or hardware configuration: Ryzen 7 3700X, 64GB RAM, bare-metal
- OS (e.g: `cat /etc/os-release`): 20.10 (Groovy Gorilla)
- Kernel (e.g. `uname -a`): Linux k01 5.8.0-45-generic #51-Ubuntu SMP Fri Feb 19 13:24:51 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm
- Network plugin and version (if this is a network-related bug): cilium 
- Others:

`Sep 29 00:09:05 k04 kubelet[1319697]: E0929 00:09:05.509324 1319697 kubelet_volumes.go:245] "There were many similar errors. Turn up verbosity to see them." err="orphaned pod \"0bee1910-1279-4f50-8b4b-765d5193a9ed\" found, but error not a directory occurred when trying to remove the volumes dir" numErrs=17
Sep 29 00:09:05 k04 systemd[1]: Removed slice libcontainer container kubepods-besteffort-pod2a93a4c4_40fa_401b_886f_5e92eeca97ae.slice.
Sep 29 00:09:06 k04 kubelet[1319697]: E0929 00:09:06.022919 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="48c12048f7e12650db9e9df927463ce13201b8e1215b373e170a68c9459e3c65"
Sep 29 00:09:06 k04 kubelet[1319697]: E0929 00:09:06.023716 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="bcf43fb78c65a7170de366ec9144ac62e189d6823548363540c875f6b9fb3231"
Sep 29 00:09:06 k04 kubelet[1319697]: E0929 00:09:06.024263 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="5074f4260947fa78be1cce16a1c55e25860fafcec6ff7f83b91f3bd919c6d2cd"
Sep 29 00:09:06 k04 kubelet[1319697]: E0929 00:09:06.024810 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="bcf43fb78c65a7170de366ec9144ac62e189d6823548363540c875f6b9fb3231"
Sep 29 00:09:06 k04 kubelet[1319697]: E0929 00:09:06.025351 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="5074f4260947fa78be1cce16a1c55e25860fafcec6ff7f83b91f3bd919c6d2cd"
Sep 29 00:09:06 k04 kubelet[1319697]: E0929 00:09:06.025887 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="48c12048f7e12650db9e9df927463ce13201b8e1215b373e170a68c9459e3c65"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.028329 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="bcf43fb78c65a7170de366ec9144ac62e189d6823548363540c875f6b9fb3231"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.029211 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="48c12048f7e12650db9e9df927463ce13201b8e1215b373e170a68c9459e3c65"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.029906 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="5074f4260947fa78be1cce16a1c55e25860fafcec6ff7f83b91f3bd919c6d2cd"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.030587 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="48c12048f7e12650db9e9df927463ce13201b8e1215b373e170a68c9459e3c65"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.031273 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="bcf43fb78c65a7170de366ec9144ac62e189d6823548363540c875f6b9fb3231"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.032011 1319697 remote_runtime.go:343] "verify ContainerStatus failed" err="status.CreatedAt is not set" containerID="5074f4260947fa78be1cce16a1c55e25860fafcec6ff7f83b91f3bd919c6d2cd"
Sep 29 00:09:07 k04 kubelet[1319697]: E0929 00:09:07.511947 1319697 kubelet_volumes.go:245] "There were many similar errors. Turn up verbosity to see them." err="orphaned pod \"0bee1910-1279-4f50-8b4b-765d5193a9ed\" found, but error not a directory occurred when trying to remove the volumes dir" numErrs=17`

**Comment by cyclinder** [comment]
https://github.com/kubernetes/kubernetes/blob/49f9e227e985bcbf6fd4781db1788b78a8e140af/pkg/kubelet/cri/remote/remote_runtime.go#L328-L344

Firstly.the error is reported by remote_runtime.go:343 that "status.CreatedAt is not set".i check out the function "verifyContainerStatus".and then i found status.CreatedAt  == 0

https://github.com/kubernetes/kubernetes/blob/49f9e227e985bcbf6fd4781db1788b78a8e140af/pkg/kubelet/cri/remote/utils.go#L78-L80

and i traced the source code in 
[https://github.com/cri-o/cri-o/blob/ec1f1cfa0a898e624feac90ebccc5101d4016754/server/container_status.go#L68-L102
](https://github.com/cri-o/cri-o/blob/ec1f1cfa0a898e624feac90ebccc5101d4016754/server/container_status.go#L68-L102
),I need to know the status of the container before I going on it.

so can you tell me what the status of the container is ？

**Comment by SergeyKanzhelev** [comment]
cc: @haircommander 

Can you take a look if there is enough information for investigation?

**Comment by haircommander** [comment]
what version of runc are you using @Jan-EC

**Comment by Jan-EC** [comment]
Hi and thanks for looking into this. 

I noticed the issue when some but not all pods created were stuck in "ContainerCreating" for many hours without any idea why and describe showed the error.

For the runc version:  

`VERSION:
   1.0.1
commit: v1.0.1-0-g4144b63
spec: 1.0.2-dev
go: go1.15.14
libseccomp: 2.4.3`

**Comment by Jan-EC** [comment]
Another update: The logs vanished when I switched the runtime to contained instead of cri-o. 
I will build a small test cluster with the same setup as before and see if the error is reproducible.

**Comment by haircommander** [comment]
what versions of cri-o were you moving between in the upgrade?

**Comment by Jan-EC** [comment]
The latest installed was

```crio version 1.21.0
Version:       1.21.0
GitCommit:     bc1ef35a932acc2f6f3b6d3eb19a4f68aa9423f6
GitTreeState:  clean
BuildDate:     2021-04-30T13:47:12Z
GoVersion:     go1.15.2
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic```

It was 1.20 before.

**Comment by haircommander** [comment]
do you happen to know the exact version of 1.20? I would like to try to reproduce

**Comment by Jan-EC** [comment]
That should have been 1.20.5 - but I cant tell for 100% sure.

**Comment by haircommander** [comment]
oops, forgot about this issue. this should have been fixed in cri-o 1.25.0 as well as the next cri-o 1.24 (1.24.4, not yet released). see https://github.com/cri-o/cri-o/pull/6122
